contracts/vaults/RibbonThetaVault.sol

```diff
@@ -318,18 +325,66 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
 
             depositReceipts[msg.sender] = VaultDeposit.DepositReceipt({
                 processed: false,
-                round: _round,
+                round: currentRound,
                 amount: uint128(newAmount)
             });
         } else {
             depositReceipts[msg.sender] = VaultDeposit.DepositReceipt({
                 processed: false,
-                round: _round,
+                round: currentRound,
                 amount: uint128(amount)
             });
         }
 
-        totalPending = totalPending.add(amount);
+        _totalPending = _totalPending.add(amount);
+
+        // If we have an unprocessed pending deposit from the previous rounds, we have to redeem it.
+        if (depositReceipt.round < currentRound && !depositReceipt.processed) {
+            _redeemDeposit(currentRound, depositReceipt);
+        }
+    }
+
+    function redeemDeposit() external nonReentrant {
+        VaultDeposit.DepositReceipt memory depositReceipt =
+            depositReceipts[msg.sender];
+        _redeemDeposit(round, depositReceipt);
+    }
+
+    function _redeemDeposit(
+        uint16 currentRound,
+        VaultDeposit.DepositReceipt memory depositReceipt
+    ) private {
+        require(!depositReceipt.processed, "Processed");
+        require(depositReceipt.round > currentRound, "Round not closed");
```

Comment:
should it be `depositReceipt.round < currentRound` based on [this](https://github.com/ribbon-finance/ribbon-v2/pull/3#discussion_r649892802)

Reply:
Yep, was a typo